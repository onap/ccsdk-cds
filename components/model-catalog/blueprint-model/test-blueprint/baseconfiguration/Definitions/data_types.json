{
  "data_types": {
    "sample-property": {
      "description": "This is sample data type",
      "version": "1.0.0",
      "properties": {
        "content": {
          "required": false,
          "type": "string"
        },
        "process-name": {
          "required": false,
          "type": "string"
        },
        "version": {
          "required": false,
          "type": "string",
          "default": "LATEST"
        }
      },
      "derived_from": "tosca.datatypes.Root"
    },
    "dt-resource-assignment-properties": {
      "description": "This is Dynamically generated data type for workflow activate",
      "version": "1.0.0",
      "properties": {
        "request-id": {
          "required": true,
          "type": "string"
        },
        "service-instance-id": {
          "required": true,
          "type": "string"
        },
        "vnf-id": {
          "required": true,
          "type": "string"
        },
        "action-name": {
          "required": true,
          "type": "string"
        },
        "scope-type": {
          "required": true,
          "type": "string"
        },
        "hostname": {
          "required": true,
          "type": "string"
        },
        "vnf_name": {
          "required": true,
          "type": "string"
        },
        "private_net_id": {
          "required": true,
          "type": "string"
        },
        "firmware-version": {
          "type" : "string"
        },
        "ip-address": {
          "type" : "string"
        }
      },
      "derived_from": "tosca.datatypes.Dynamic"
    },
    "dt-assign-activate-properties": {
      "description": "This is Dynamically generated data type for workflow assign-activate",
      "version": "1.0.0",
      "properties": {
        "request-id": {
          "required": true,
          "type": "string"
        },
        "action-name": {
          "required": true,
          "type": "string"
        },
        "scope-type": {
          "required": true,
          "type": "string"
        },
        "hostname": {
          "required": true,
          "type": "string"
        }
      },
      "derived_from": "tosca.datatypes.Dynamic"
    }
  }
}