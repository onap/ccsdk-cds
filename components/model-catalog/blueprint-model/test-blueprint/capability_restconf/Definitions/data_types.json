{
  "data_types" : {
    "dt-config-assign-properties" : {
      "description" : "Dynamic DataType definition for workflow(config-assign).",
      "version" : "1.0.0",
      "properties" : { },
      "derived_from" : "tosca.datatypes.Dynamic"
    },
    "dt-config-deploy-properties" : {
      "description" : "Dynamic DataType definition for workflow(config-deploy).",
      "version" : "1.0.0",
      "properties" : {
        "pnf-id" : {
          "type" : "string"
        },
        "pnf-name" : {
          "type" : "string"
        }
      },
      "derived_from" : "tosca.datatypes.Dynamic"
    }
  }
}