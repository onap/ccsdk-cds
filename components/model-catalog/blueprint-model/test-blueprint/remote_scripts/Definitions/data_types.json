{
  "data_types": {
    "dt-system-packages": {
      "description": "This represent System Package Data Type",
      "version": "1.0.0",
      "properties": {
        "type": {
          "required": true,
          "type": "string",
          "constraints": [
            {
              "valid_values": [
                "ANSIBLE", "PYTHON"
              ]
            }
          ]
        },
        "package": {
          "required": true,
          "type": "list",
          "entry_schema" : {
            "type" : "string"
          }
        }
      },
      "derived_from": "tosca.datatypes.Root"
    }
  }
}