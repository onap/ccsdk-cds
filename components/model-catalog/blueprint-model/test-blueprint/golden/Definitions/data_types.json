{
  "data_types" : {
    "dt-config-assign-properties" : {
      "description" : "Dynamic DataType definition for workflow(config-assign).",
      "version" : "1.0.0",
      "properties" : {
        "hostname" : {
          "type" : "string"
        },
        "vfw_interface_ip" : {
          "type" : "dt-netbox-ip"
        },
        "interface-name" : {
          "type" : "string"
        },
        "golden-vf-module-type" : {
          "type" : "string"
        },
        "interface-description" : {
          "type" : "string"
        },
        "unit-number" : {
          "type" : "string"
        },
        "prefix-id" : {
          "type" : "string",
          "default" : "6"
        },
        "vf-module-number" : {
          "type" : "string",
          "default" : "3"
        }
      },
      "derived_from" : "tosca.datatypes.Dynamic"
    },
    "dt-netbox-ip" : {
      "description" : "This is Netbox IP Data Type",
      "version" : "1.0.0",
      "properties" : {
        "address" : {
          "required" : true,
          "type" : "string"
        },
        "id" : {
          "required" : true,
          "type" : "integer"
        }
      },
      "derived_from" : "tosca.datatypes.Root"
    },
    "dt-resource-assignment-properties" : {
      "description" : "Dynamic DataType definition for workflow(resource-assignment).",
      "version" : "1.0.0",
      "properties" : {
        "vfw_interface_ip" : {
          "type" : "dt-netbox-ip"
        },
        "interface-description" : {
          "type" : "string"
        },
        "vf-module-type" : {
          "type" : "string"
        },
        "vf-module-number" : {
          "type" : "string",
          "default" : "3"
        },
        "prefix-id" : {
          "type" : "integer",
          "default" : 6
        }
      },
      "derived_from" : "tosca.datatypes.Dynamic"
    },
    "dt-rollback-properties" : {
      "description" : "Dynamic DataType definition for workflow(rollback).",
      "version" : "1.0.0",
      "properties" : { },
      "derived_from" : "tosca.datatypes.Dynamic"
    }
  }
}